### ISSUE TITLE
ktlintFormat does not autoformat

### ISSUE THREAD

**Issue by vanniktech** [shyiko/ktlint#129]
When I apply this section into my `app/build.gradle` file:

```groovy
task ktlintFormat(type: JavaExec, group: "formatting") {
  description = "Fix Kotlin code style deviations."
  classpath = configurations.ktlint
  main = "com.github.shyiko.ktlint.Main"
  args "-F", "src/**/*.kt"
}
```

and I'm running it via `./gradlew ktlintFormat` I'd expect to get auto corrected files. The only thing I'm seeing in the console is that the warnings are reported but nothing gets fixed. Is there any flag missing for the executable?

I'm using 0.14.0

**Comment by shyiko** [comment]
@vanniktech ktlint@0.14.0 --format reports errors which it cannot fix automatically. As of 0.14.0 this includes `max-line-length`, `no-wildcard-imports` and `indent`. If you are seeing a lint error outside of this list please let me know. :bowing_man:

**Comment by vanniktech** [comment]
Oh it's indent. Do we have an issue to autofix these?

**Comment by KolesnichenkoDS** [comment]
@shyiko I cannot make it even report indenting errors - it just ignores wrong indentation. Other types of errors (max line length, wildcard imports etc) are detected.

**Comment by KolesnichenkoDS** [comment]
And when do you plan to implement autoformatting in ktlint?

**Comment by shyiko** [comment]
@KolesnichenkoDS a week after the New Year Eve.

**Comment by KolesnichenkoDS** [comment]
@shyiko cool!

**Comment by dannyroa** [comment]
Is indenting on autoformat working now? It's just getting reported when I tried it.

**Comment by shyiko** [comment]
@dannyroa partially: parameter list wrapping auto-correction included in 0.16.0+; general indentation is in the works.

**Comment by dannyroa** [comment]
@shyiko: Any way I can help build this feature? Thanks.

**Comment by shyiko** [comment]
@dannyroa That would be great! 

@MyDogTom made a good progress in https://github.com/shyiko/ktlint/pull/137 (changes are in develop branch) but it needs more work;  
(code indented like show below is not corrected (at all)) 
```kotlin
fun a() {
class A() {
val a = 1
}
}
```

I feel like instead of calculating previous indent on each [visit(...)](https://github.com/shyiko/ktlint/blob/master/ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/IndentationRule.kt#L28) we should iterate through AST [once](https://github.com/shyiko/ktlint/blob/master/ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/ParameterListWrappingRule.kt#L118) ([on FILE](https://github.com/shyiko/ktlint/blob/master/ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/IndentationRule.kt#L33)) and report/correct wrong indents are we go (based on the nesting so far).

(current rule sources are here - https://github.com/shyiko/ktlint/blob/master/ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/IndentationRule.kt)

P.S. You might find https://github.com/shyiko/ktlint#ast helpful.

**Comment by MyDogTom** [comment]
@shyiko Code you mentioned is not formatted, because rule is not strict enough. Violation is just not detected. I've explained it in details [here](https://github.com/shyiko/ktlint/issues/141#issuecomment-361203788). It's not related to the calculation of previous intent. Rule just should be more strict.
But I agree that instead of visiting each node, we could go through entire file only once and just persist previous indent. That should be more efficient.

**Comment by shyiko** [comment]
@MyDogTom https://github.com/shyiko/ktlint/issues/141#issuecomment-361203788 << right.
